frontend_tests/node_tests/filter.js

```diff
@@ -108,7 +112,32 @@ run_test('basics', () => {
     assert(filter.has_operator('has'));
     assert(!filter.can_apply_locally());
 });
+run_test('show_first_unread', () => {
+    var operators = [
+        {operator: 'is', operand: 'any'},
+    ];
+    var filter = new Filter(operators);
+    assert(filter.allow_use_first_unread_when_narrowing());
+
+    operators = [
+        {operator: 'search', operand: 'query to search'},
+    ];
+    filter = new Filter(operators);
+    assert(!filter.allow_use_first_unread_when_narrowing());
 
+    filter = new Filter();
+    filter.can_mark_messages_read = () => true;
+    assert(filter.allow_use_first_unread_when_narrowing());
+
+    // Side case
+    operators = [
+        {operator: 'is', operand: 'any'},
+    ];
+    filter = new Filter(operators);
+    filter.can_mark_messages_read = () => false;
+    assert(filter.allow_use_first_unread_when_narrowing());
```

Comment:
Is this test case correct ? I was a bit confused, for eg. if a user is searching in is:mentioned narrow. Should we see unread message first in view ?

Reply:
@timabbott Is this test case correct ? I was a bit confused, for eg. if a user is searching in is:mentioned narrow. Should we see unread message first in view ?